streams/src/test/java/org/apache/kafka/streams/kstream/RepartitionTopicNamingTest.java

```diff
@@ -0,0 +1,544 @@
+        assertTrue(unOptimizedTopology.contains(secondRepartitionTopicName + "-repartition"));
+        assertTrue(unOptimizedTopology.contains(thirdRepartitionTopicName + "-repartition"));
+        assertTrue(unOptimizedTopology.contains(fourthRepartitionTopicName + "-left-repartition"));
+
+    }
+
+    @Test
+    public void shouldKeepRepartitionTopicNameForJoins() {
+
+        final String expectedLeftRepartitionTopic = "(topic: my-join-left-repartition)";
+        final String expectedRightRepartitionTopic = "(topic: my-join-right-repartition)";
+
+
+        final String joinTopologyFirst = buildStreamJoin(false);
+
+        assertTrue(joinTopologyFirst.contains(expectedLeftRepartitionTopic));
+        assertTrue(joinTopologyFirst.contains(expectedRightRepartitionTopic));
+
+        final String joinTopologyUpdated = buildStreamJoin(true);
+
+        assertTrue(joinTopologyUpdated.contains(expectedLeftRepartitionTopic));
+        assertTrue(joinTopologyUpdated.contains(expectedRightRepartitionTopic));
+    }
+
+    @Test
+    public void shouldKeepRepartitionTopicNameForGroupByKeyTimeWindows() {
+
+        final String expectedTimeWindowRepartitionTopic = "(topic: time-window-grouping-repartition)";
+
+        final String timeWindowGroupingRepartitionTopology = buildStreamGroupByKeyTimeWindows(false, true);
+        assertTrue(timeWindowGroupingRepartitionTopology.contains(expectedTimeWindowRepartitionTopic));
+
+        final String timeWindowGroupingUpdatedTopology = buildStreamGroupByKeyTimeWindows(true, true);
+        assertTrue(timeWindowGroupingUpdatedTopology.contains(expectedTimeWindowRepartitionTopic));
+    }
+
+    @Test
+    public void shouldKeepRepartitionTopicNameForGroupByTimeWindows() {
+
+        final String expectedTimeWindowRepartitionTopic = "(topic: time-window-grouping-repartition)";
+
+        final String timeWindowGroupingRepartitionTopology = buildStreamGroupByKeyTimeWindows(false, false);
+        assertTrue(timeWindowGroupingRepartitionTopology.contains(expectedTimeWindowRepartitionTopic));
+
+        final String timeWindowGroupingUpdatedTopology = buildStreamGroupByKeyTimeWindows(true, false);
+        assertTrue(timeWindowGroupingUpdatedTopology.contains(expectedTimeWindowRepartitionTopic));
+    }
+
+
+    @Test
+    public void shouldKeepRepartitionTopicNameForGroupByKeyNoWindows() {
+
+        final String expectedNoWindowRepartitionTopic = "(topic: kstream-grouping-repartition)";
+
+        final String noWindowGroupingRepartitionTopology = buildStreamGroupByKeyNoWindows(false, true);
+        assertTrue(noWindowGroupingRepartitionTopology.contains(expectedNoWindowRepartitionTopic));
+
+        final String noWindowGroupingUpdatedTopology = buildStreamGroupByKeyNoWindows(true, true);
+        assertTrue(noWindowGroupingUpdatedTopology.contains(expectedNoWindowRepartitionTopic));
+    }
+
+    @Test
+    public void shouldKeepRepartitionTopicNameForGroupByKeyWindows() {
```

Comment:
Should this be `shouldKeepRepartitionTopicNameForGroupByKeyNoWindows`?

Reply:
good catch actually `shouldKeepRepartitionTopicNameForGroupByNoWindows`